src/services/fileStorage/utils/filePathHelper.js

```diff
@@ -1,9 +1,32 @@
 const removeLeadingSlash = (path) => (path[0] === '/' ? path.substring(1) : path);
 
-const generateFileNameSuffix = (fileName) => {
+const s3FileNameFilterExpression = /[^0-9A-Za-z\-_.]+/g;
+/**
+ * ensures a valid filename for s3 removing invalid characters from local/internal filename
+ * @param {*} fileName
+ * @returns
+ */
+const whitelistFileName = (fileName) => {
+	// see https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-keys.html for valid characters
+	// the regex below reduces allowed special chars to hyphen, underscore, period
+	// multiple invalid occurrences are all replaced by one single '-'
+	// ab3&s --> ab3-s
+	// db/&%$s --> db-s
 	// eslint-disable-next-line no-useless-escape
-	const newString = fileName.replace(/[';:=#*+\[\]~<{\\()}>§$%&|^£@±?!"`□\s]/g, '-');
-	return `${Date.now()}-${newString.toLowerCase()}`;
+	const lowerCaseFileName = fileName.toLowerCase();
+	const result = lowerCaseFileName.replace(s3FileNameFilterExpression, '-');
+	return result;
+};
+
+/**
+ * adds a timestamp prefix to filename and ensures maxlength of 1024 characters
+ */
+const generateFileNameSuffix = (fileName) => {
```

Comment:
Is this really a suffix filter now, maybe we must rename it to prevent confusing? 
It is only used for copy files.

Reply:
it's adding a timestamp, shorten the string and removes invalid characters